pkg/cache/v3/snapshot.go

```diff
@@ -266,3 +269,30 @@ func (s *Snapshot) ConstructVersionMap() error {
 
 	return nil
 }
+
+// Respond to a delta watch with the provided snapshot value. If the response is nil, there has been no state change.
+func (s *Snapshot) respondDelta(ctx context.Context, request *DeltaRequest, value chan DeltaResponse, state stream.StreamState, log log.Logger) (*RawDeltaResponse, error) {
```

Comment:
I'm cool with the code duplication. They're different cache implementations so it makes sense for them to be localized anyways. It seems a bit easier to understand instead of trying to tie the two together